lib/xhr/XHRShim.js

```diff
@@ -0,0 +1,142 @@
+/*jslint vars: true, plusplus: true, devel: true, nomen: true, indent: 4, forin: true, maxerr: 50, regexp: true */
+/* global DOMParser */
+(function(global) {
+    "use strict";
+
+    // This script requires access to the transport to send
+    // file requests to the editor as commands
+    var transport = global._Brackets_LiveDev_Transport;
+
+    var XMLHttpRequest = global.XMLHttpRequest;
+    if(!XMLHttpRequest) {
+        return;
+    }
+
+    function sendMessage(msg) {
+        if(!transport || !transport.send) {
```

Comment:
There's already code to deal with not having `transport.send` in Brackets, I'd just check on the value of `transport`